include/eve/detail/assert_utils.hpp

```diff
@@ -14,7 +14,7 @@
 namespace eve::detail
 {
   template<typename A0, eve::value A1>
-  constexpr EVE_FORCEINLINE bool assert_good_shift(A1 const &t) noexcept
+  constexpr EVE_FORCEINLINE bool assert_good_shift(A1 t) noexcept
```

Comment:
I find it weird that we allow a shift by negative number for signed value type.

gcc complains: https://godbolt.org/z/hEdT5zfed

Not for this pr though.

Reply:
You either go all signed shifts or none signed shifts. The "signed only for signed numbers" makes no sense.